### ISSUE TITLE
Web interface ideas

### ISSUE THREAD

**Issue by mark66j** [jmlzone/PiPtr#20]
Here are some thoughts about web interfaces for the system. This is written without much knowledge of the code so far so it may not entirely make sense, but anyway:

1) I think each controller should have a simple web interface, maybe only accessible to admins, to show the status of that board and allow changes
2) A central web page could show the entire Minuteman system status (somewhat like the current page, but maybe adding a map view). This would be view-only for the public, but admins would be able to click on a repeater and issue commands.
3) Because of browser security rules, the web page may not be able to send commands directly to the repeaters, so a central server may be needed to do this. 
4) I am thinking that each controller could have a web services interface (maybe REST?) for use by this central page.
5) It may be helpful to have a simpler central control page that is easier to use on a smart phone.
6) Authentication is tricky -- connections to each controller would need to be protected, but how to push changes to all devices in the system? RADIUS is the traditional way here but there might be better choices.

**Comment by jmlzone** [comment]
There is a need for a consolidated control op system overview where a control up can have a button like link all to hub 1 or link these to hub 2 etc.  This may require some sort of middle server code that then breaks the meta command into commands to each repeater as needed.

However I agree a consolidated system status public view is also great!  Some of this can be hosted on MMRA.org and would be part of MMR specific code, while the generic per repeater code (which could even be hosted on the Pi's them selves) would be more a direct part of the public project.

**Comment by mark66j** [comment]
I agree. The public project code could expose a web services API that MMRA would use for the central control system and to provide the status view. Organizations that have multiple repeaters could write their own central control, maybe using the MMRA code as a template.
